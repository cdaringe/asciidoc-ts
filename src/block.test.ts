import { describe, expect, test } from "vitest";
import { toAST } from "./mod.js";

test("Parse block with metadata", async ({ expect }) => {
  const input = `.Block Title
[#id, .role]
This is a paragraph with metadata.`.trimStart();
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});

test("Parse paragraph block", async ({ expect }) => {
  const input = `This is a simple paragraph block.
It can span multiple lines.`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});

test("Parse header block", async ({ expect }) => {
  const input = `== Section Header`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});

test("Parse listing block", async ({ expect }) => {
  const input = `[source,javascript]
----
function hello() {
  console.log("Hello, world!");
}
----`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});

test.skip("Parse block quote", async ({ expect }) => {
  const input = `[quote, Albert Einstein]
____
The important thing is not to stop questioning. Curiosity has its own reason for existing.
____`;
  const result = toAST(input);
  expect(result).toMatchSnapshot();
});
