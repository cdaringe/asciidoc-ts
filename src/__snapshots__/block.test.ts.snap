// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Parse block with metadata 1`] = `
{
  "blocks": [
    {
      "content": [
        {
          "content": "This is a paragraph with metadata.",
          "type": "PlainText",
        },
      ],
      "metadata": {
        "attributes": [
          {
            "name": "#id",
            "type": "AttributeEntry",
          },
          {
            "name": ".role",
            "type": "AttributeEntry",
          },
        ],
        "title": "Block Title",
      },
      "type": "Paragraph",
    },
  ],
  "type": "Document",
}
`;

exports[`Parse header block 1`] = `
{
  "blocks": [
    {
      "content": [
        {
          "content": "Section Header",
          "type": "PlainText",
        },
      ],
      "level": 2,
      "type": "Header",
    },
  ],
  "type": "Document",
}
`;

exports[`Parse listing block 1`] = `
{
  "blocks": [
    {
      "content": "function hello() {
  console.log("Hello, world!");
}",
      "delimited": true,
      "metadata": {
        "attributes": [
          {
            "name": "source",
            "type": "AttributeEntry",
          },
          {
            "name": "javascript",
            "type": "AttributeEntry",
          },
        ],
      },
      "type": "CodeBlock",
    },
  ],
  "type": "Document",
}
`;

exports[`Parse paragraph block 1`] = `
{
  "blocks": [
    {
      "content": [
        {
          "content": "This is a simple paragraph block.",
          "type": "PlainText",
        },
        {
          "content": "It can span multiple lines.",
          "type": "PlainText",
        },
      ],
      "type": "Paragraph",
    },
  ],
  "type": "Document",
}
`;
